topics/engineering/improving-wiki-rif-search-in-genenetwork.gmi

```diff
@@ -0,0 +1,117 @@
+# Improving RIF+WIKI Search
+
+* author: bonfacem
+* reviewed-by: jnduli
+
+At the time of this writing, WIKI and/or RIF Search is extremely slow for MySQL.  As an example, searching: "WIKI=nicotine MEAN=(12.103 12.105)" causes an Nginx time-out in Genenetwork2.  This blog discusses how we improved the WIKI+RIF search using XAPIAN and some of our key learnings.
+
+### TLDR; Key Learnings from Adding RIF+WIKI to the Index
+
+* Compacting is IO bound.
+* Instrument your indexing script and appropriately choose an appropriate cpu_count that fits your needs.
```

Comment:
`Test the indexing script to choose an appropriate parallel process count. In our case, we chose one that didn't use too much RAM`

Key is that we can choose cpu_count but rather the no of parallel processes